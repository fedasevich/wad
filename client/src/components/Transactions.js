import React, { useContext, useEffect, useMemo, useState } from 'react'
import {observer} from 'mobx-react-lite'
import { Button, ButtonGroup, ButtonToolbar, Col, Container, Row } from 'react-bootstrap';
import { Context } from '../index';
import { changeTransaction, deleteTransaction, fetchTransaction } from '../http/transactionApi';
import Modal from './modal/modal';
import { runInAction } from 'mobx';
import { Icons } from '../ui/Icons/CategoryIcons/CategoryIcons';
import ChangeTransactionModal from './ChangeTransactionModal';



const Transactions = observer(() => {
  const {category,wallet} = useContext(Context) 
  const [buttonVisible,setButtonVisible] = useState(true)
  const [changeTransactionModal,setChangeTransactionModal] = useState(false)
  const [transactionsSort,setTransactionsSort] = useState("DESC")
  const [newSum,setNewSum] = useState('')
  const [newDescription,setNewDescription] = useState('')
  useEffect(()=>{
    try {
      fetchTransaction(category.transactionsPage,category.transactionsLimit,transactionsSort).
      then(data=> {category.setTransactions(data.rows)    
       
    })
    } catch(e) {
      alert(e.response.data.message);
    }
   

  },[category.transactionsLimit,transactionsSort,category])


 
const data = category.transactions

  const transactions = data.reduce((transactions, transactionItem) => {
    console.log("re")
    const date = transactionItem.createdAt.split('T')[0];
    if (!transactions[date]) {
      transactions[date] = [];
    }
    transactions[date].push(transactionItem);
    return transactions;
  }, {})
  
  const transactionArrays = Object.keys(transactions).map((date) => {
    console.log("rere")
    return {
      date,
      trs: transactions[date]
    };
  })

  console.log({...transactionArrays})

  const getIconIdFromCategoryById =(id)=> {
  
    console.log("icon")
    let findCategory = category.categories.find((category)=> category.id === id)
    return findCategory?.iconId
  }

  return (
   <>
   <Container> 
    <h2>Transactions:</h2>
 
<Button onClick={()=> {
setTransactionsSort(transactionsSort==="DESC" ? "ASC" : "DESC" )
category.setTransactionsPage(1)
setButtonVisible(true)

}}>sort</Button>
<ButtonToolbar className="justify-content-end"aria-label="Toolbar with button groups">
      <ButtonGroup className="me-2" aria-label="First group">
      <Button onClick={()=>{
      category.setTransactionsLimit(5);
      category.setTransactionsPage(1)
      setButtonVisible(true)
    }}>5</Button>
    <Button onClick={()=>{
      category.setTransactionsLimit(10);
      category.setTransactionsPage(1)
      setButtonVisible(true)
    }}>10</Button>
    <Button onClick={()=>{
      category.setTransactionsLimit(20);
      category.setTransactionsPage(1)
      setButtonVisible(true)
    }}>20</Button>
    <Button onClick={()=>{
      category.setTransactionsLimit(30);
      category.setTransactionsPage(1)
      setButtonVisible(true)
    }}>30</Button>
      </ButtonGroup>
    </ButtonToolbar>




{transactionArrays.map(transactionsMap=>

  <Row>
    
     
     <h2 key={transactionsMap.date}>{transactionsMap.date}</h2> 
   {transactionsMap.trs.map(transactions=>

<Col key={transactions.id} md="12" className="d-inline-flex justify-content-between">
         <Icons iconId={getIconIdFromCategoryById(transactions.categoryId)}></Icons> 
       <h2>{transactions.description}</h2>
       {console.log("map")}
    <div className="d-flex flex-direction-row">  
      <h4>{transactions.sum}</h4>
       <Button className="ml-2 btn-danger" onClick={()=>{
          
        
       try {
        deleteTransaction(transactions.id).
        then(()=> {
          runInAction(()=>{
            const transactionIndex =  category.transactions.findIndex(transaction => transaction.id === transactions.id)
        
            const categoryIndex = category.categories.findIndex(category =>category.id === transactions.categoryId)
            if(transactions.walletId !== -1) 
           {
             const walletIndex = wallet.wallets.findIndex(wallet => wallet.id === transactions.walletId)
            wallet.wallets[walletIndex].balance += parseFloat(category.transactions[transactionIndex].sum)
          }
            category.categories[categoryIndex].spent -= parseFloat(category.transactions[transactionIndex].sum)
         
         category.transactions.splice(transactionIndex, 1)
         category.setTransactions(category.transactions)
         setNewSum('')
         setNewDescription("")
          })
         
        })
      } catch(e) {
        alert(e.response.data.message);
      }
      }}> Delete</Button>
       <Button onClick={()=>{
        category.setSelectedTransaction(transactions)
        console.log(category.selectedTransaction.categoryId)
        setChangeTransactionModal(true)
       }}>Change</Button>
       </div> 
      </Col>
     
     )}
     </Row>
    
    )}
  
{ buttonVisible ?    
    <Button className={"d-flex justify-content-center mb-5"} onClick={()=> {
      category.setTransactionsPage(category.transactionsPage+1);  
      console.log(category.transactionsPage)
      try {
        fetchTransaction(category.transactionsPage,category.transactionsLimit,transactionsSort).then(data=> 
          {
            if(!data.rows.length) {
              setButtonVisible(false)
            };
        runInAction(()=> {
          category.transactions.push(...data.rows)
        }) 
          
        
        

          })
      } catch(e) {
        alert(e.response.data.message);
      }

      }}>Load More</Button>:
      <h2 className="text-center">There is nothing to load...</h2>
      }
         </Container>

      
          <ChangeTransactionModal changeTransactionModal={changeTransactionModal} setChangeTransactionModal={setChangeTransactionModal} category={category} wallet={wallet}/>
     

      </>
  );

});

export default Transactions