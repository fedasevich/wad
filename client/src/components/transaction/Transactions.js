import React, { useContext, useEffect, useMemo, useState } from 'react'
import { observer } from 'mobx-react-lite'
import { Accordion, Button, ButtonGroup, ButtonToolbar, Col, Container, Row, useAccordionButton } from 'react-bootstrap';
import { Context } from '../../index';
import { changeTransaction, deleteTransaction, fetchTransaction } from '../../http/transactionApi';
import Modal from '../modal/modal';
import { runInAction } from 'mobx';
import { Icons } from '../../ui/Icons/CategoryIcons/CategoryIcons';
import ChangeTransactionModal from './ChangeTransactionModal';
import TransactionProvider from './TransactionProvider';
import { format, formatISO, parseISO } from 'date-fns';





// const Transactions = observer(() => {
//   const {category,wallet} = useContext(Context) 
//   const [buttonVisible,setButtonVisible] = useState(true)
//   const [changeTransactionModal,setChangeTransactionModal] = useState(false)

//   const [newSum,setNewSum] = useState('')
//   const [newDescription,setNewDescription] = useState('')
//   useEffect(()=>{
//     try {
//       fetchTransaction(category.transactionsPage,category.transactionsLimit,category.transactionsSort).
//       then(data=> {category.setTransactions(data.rows)    

//     })
//     } catch(e) {
//       alert(e.response.data.message);
//     }


//   },[category.transactionsLimit,category.transactionsSort,category])



// const data = category.transactions

//   const transactions = data.reduce((transactions, transactionItem) => {

//     const date = transactionItem.createdAt.split('T')[0];
//     if (!transactions[date]) {
//       transactions[date] = [];
//     }
//     transactions[date].push(transactionItem);
//     return transactions;
//   }, {})

//   const transactionArrays = Object.keys(transactions).map((date) => {

//     return {
//       date,
//       trs: transactions[date]
//     };
//   })



//   const getIconIdFromCategoryById =(id)=> {

//     let findCategory = category.categories.find((category)=> category.id === id)
//     return findCategory?.iconId
//   }

//   return (
//    <>
//    <Container> 
//     <h2>Transactions:</h2>

// <Button onClick={()=> {
// category.setTransactionsSort(category.transactionsSort==="DESC" ? "ASC" : "DESC" )

// category.setTransactionsPage(1)
// setButtonVisible(true)

// }}>sort</Button>
// <ButtonToolbar className="justify-content-end"aria-label="Toolbar with button groups">
//       <ButtonGroup className="me-2" aria-label="First group">
//       <Button onClick={()=>{
//       category.setTransactionsLimit(5);
//       category.setTransactionsPage(1)
//       setButtonVisible(true)
//     }}>5</Button>
//     <Button onClick={()=>{
//       category.setTransactionsLimit(10);
//       category.setTransactionsPage(1)
//       setButtonVisible(true)
//     }}>10</Button>
//     <Button onClick={()=>{
//       category.setTransactionsLimit(20);
//       category.setTransactionsPage(1)
//       setButtonVisible(true)
//     }}>20</Button>
//     <Button onClick={()=>{
//       category.setTransactionsLimit(30);
//       category.setTransactionsPage(1)
//       setButtonVisible(true)
//     }}>30</Button>
//       </ButtonGroup>
//     </ButtonToolbar>




// {transactionArrays.map((transactionsMap,index)=>

//   <Row  key={index}>


//      <h2>{transactionsMap.date}</h2> 
//    {transactionsMap.trs.map(transactions=>

// <Col key={transactions.id} md="12" className="d-inline-flex justify-content-between">
//          <Icons iconId={getIconIdFromCategoryById(transactions.categoryId)}></Icons> 
//        <h2>{transactions.description}</h2>

//     <div className="d-flex flex-direction-row">  
//       <h4>{transactions.sum}</h4>
//        <Button className="ml-2 btn-danger" onClick={()=>{
//           category.deleteTransaction(transactions.id,transactions.categoryId,transactions.walletId,wallet)

//       }}> Delete</Button>
//        <Button onClick={()=>{
//         category.setSelectedTransaction(transactions)

//         setChangeTransactionModal(true)
//        }}>Change</Button>
//        </div> 
//       </Col>

//      )}
//      </Row>

//     )}

// { buttonVisible ?    
//     <Button className={"d-flex justify-content-center mb-5"} onClick={()=> {
//       category.setTransactionsPage(category.transactionsPage+1);  

//       try {
//         fetchTransaction(category.transactionsPage,category.transactionsLimit,category.transactionsSort).then(data=> 
//           {
//             if(!data.rows.length) {
//               setButtonVisible(false)
//             };
//         runInAction(()=> {
//           category.transactions.push(...data.rows)
//         }) 




//           })
//       } catch(e) {
//         alert(e.response.data.message);
//       }

//       }}>Load More</Button>:
//       <h2 className="text-center">There is nothing to load...</h2>
//       }
//          </Container>


//           {/* <ChangeTransactionModal changeTransactionModal={changeTransactionModal} setChangeTransactionModal={setChangeTransactionModal} category={category} wallet={wallet}/>
//       */}

//       </>
//   );

// });

function TransactionToggle({ children, eventKey }) {
  const decoratedOnClick = useAccordionButton(eventKey);

  return (
    <div

      onClick={decoratedOnClick}
    >
      {children}
    </div>
  );
}

const Transactions = observer(({ actions }) => {
  const { category, wallet } = useContext(Context)
  const [buttonVisible, setButtonVisible] = useState(true)
  const [changeTransactionModal, setChangeTransactionModal] = useState(false)
  const [newSum, setNewSum] = useState('')
  const [newDescription, setNewDescription] = useState('')
  useEffect(() => {
    try {
      fetchTransaction(category.transactionsPage, category.transactionsLimit, category.transactionsSort).
        then(data => {
          category.setTransactions(data.rows)

        })
    } catch (e) {
      alert(e.response.data.message);
    }


  }, [category.transactionsLimit, category.transactionsSort, category])



  const data = category.transactions

  console.log(data)
  const transactions = data.reduce((transactions, transactionItem) => {
    const date = format(parseISO(transactionItem.createdAt), "d MMMM y");
    if (!transactions[date]) {
      transactions[date] = [];
    }
    transactions[date].push(transactionItem);
    return transactions;
  }, {})

  const transactionArrays = Object.keys(transactions).map((date) => {

    return {
      date,
      trs: transactions[date]
    };
  })





  const loadMoreTransactions = () => {
    category.setTransactionsPage(category.transactionsPage + 1);

    try {
      fetchTransaction(category.transactionsPage, category.transactionsLimit, category.transactionsSort).then(data => {
        if (!data.rows.length) {
          setButtonVisible(false)
        };
        runInAction(() => {
          category.transactions.push(...data.rows)
        })




      })
    } catch (e) {
      alert(e.response.data.message);
    }
  }







  return (
    <>

      <TransactionProvider>
        {actions && <TransactionProvider.Actions setButtonVisible={setButtonVisible} category={category} />}

        {transactionArrays.map((transactionsMap, index) => {
          return (


            <Row key={transactionsMap.date}>
              <TransactionProvider.Transaction.Date key={index} date={transactionsMap.date} />
              <Accordion>
                {transactionsMap.trs.map((transactions, index) => {

                  return (
                    <TransactionToggle eventKey={index} key={transactions.id}>
                      <TransactionProvider.Transaction transaction={transactions} wallet={wallet} category={category} index={index} setChangeTransactionModal={setChangeTransactionModal} />
                    </TransactionToggle>

                  )
                }

                )}
              </Accordion>

            </Row>

          )

        }

        )}

        <Row>
          <TransactionProvider.LoadMore buttonVisible={buttonVisible}
            setButtonVisible={setButtonVisible}
            fetchTransaction={loadMoreTransactions}
          />
        </Row>

        <ChangeTransactionModal changeTransactionModal={changeTransactionModal.active} id={changeTransactionModal.id} setChangeTransactionModal={setChangeTransactionModal} />
      </TransactionProvider>
    </>

  );

});

export default Transactions