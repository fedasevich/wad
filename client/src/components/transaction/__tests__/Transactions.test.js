import '@testing-library/jest-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { mockTransactions } from '../../../../setupTests';
import Transactions from '../Transactions';

jest.mock('../../../store');
jest.mock('../ChangeTransactionModal', () => jest.fn(() => null));
jest.mock('../../modal/modal');
const mockHandleFetchTransactions = jest.fn();

describe('Transactions.js', () => {
  beforeEach(() => {
    mockHandleFetchTransactions.mockResolvedValue({ rows: mockTransactions });
  });

  it('renders the component with transactions', () => {
    render(<Transactions handleFetchTransactions={mockHandleFetchTransactions} />);

    expect(
      screen.getByRole('button', {
        name: /load more/i
      })
    ).toBeInTheDocument();
  });

  it('has nothing to load when button is clicked', async () => {
    const mockFetchTransactions = jest.fn().mockResolvedValue({ rows: [] });

    render(<Transactions handleFetchTransactions={mockFetchTransactions} />);

    const loadMoreButton = screen.getByRole('button', {
      name: /load more/i
    });

    await act(() => {
      fireEvent.click(loadMoreButton);
    });

    await waitFor(() => {
      expect(mockFetchTransactions).toHaveBeenCalledTimes(2);

      expect(loadMoreButton).not.toBeInTheDocument();
      expect(
        screen.getByRole('heading', {
          name: /there is nothing to load\.\.\./i
        })
      ).toBeInTheDocument();
    });
  });

  it('loads more transactions when button is clicked', async () => {
    const mockFetchTransactions = jest
      .fn()
      .mockResolvedValue({ rows: [{ ...mockTransactions[0], id: 1991, description: 'Mock Transaction' }] });

    render(<Transactions handleFetchTransactions={mockFetchTransactions} />);
    const loadMoreButton = screen.getByRole('button', {
      name: /load more/i
    });
    await act(() => {
      fireEvent.click(loadMoreButton);
    });

    expect(mockFetchTransactions).toHaveBeenCalledTimes(2);

    expect(loadMoreButton).toBeInTheDocument();
  });

  it('renders transactions according to snapshot', async () => {
    const { container } = await act(() =>
      render(<Transactions handleFetchTransactions={mockHandleFetchTransactions} />)
    );
    expect(container).toMatchSnapshot();
  });
});
