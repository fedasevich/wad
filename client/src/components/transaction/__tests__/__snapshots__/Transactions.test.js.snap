// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions.js renders transactions according to snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      23 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Leisure
                  </h5>
                </div>
                <h5>
                  -
                  1.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Mock Transaction
                  </h5>
                </div>
                <h5>
                  -
                  1.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      20 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Family
                  </h5>
                </div>
                <h5>
                  -
                  1.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      16 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Groceries
                  </h5>
                </div>
                <h5>
                  -
                  2.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      15 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Gifts
                  </h5>
                </div>
                <h5>
                  -
                  3.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Groceries
                  </h5>
                </div>
                <h5>
                  -
                  10.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      14 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Groceries
                  </h5>
                </div>
                <h5>
                  -
                  2.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Groceries
                  </h5>
                </div>
                <h5>
                  -
                  1.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      12 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    фыв
                  </h5>
                </div>
                <h5>
                  -
                  20.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <h4
      class="fw-medium"
    >
      25 August 2023
    </h4>
    <div
      class="accordion"
    >
      <div
        class="d-inline-flex justify-content-between my-1 w-100 col-md-12"
      >
        <div
          class="border-0 w-100  card"
        >
          <div
            class="bg-none px-0 px-sm-3 d-flex w-100 justify-content-between border-0 card-header"
          >
            <div
              class="w-100 cursor-pointer"
              role="button"
              tabindex="0"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-between"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  <div
                    class="bg-main-blue component-one-third-border-radius categoryIcon"
                    style="background-color: rgb(250, 8, 52);"
                  >
                    <svg>
                      palette.svg
                    </svg>
                  </div>
                  <h5
                    class="ms-3 mb-0"
                  >
                    Family
                  </h5>
                </div>
                <h5>
                  -
                  20.00
                   
                  ₴
                </h5>
              </div>
            </div>
          </div>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="d-flex justify-content-evenly bg-light-blue text-light component-one-third-border-radius-bottom card-body"
            >
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  TrashCircle.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Delete
                </span>
              </button>
              <button
                class="cursor-pointer"
                tabindex="0"
                type="button"
              >
                <svg>
                  Icon-1.svg
                </svg>
                <span
                  class="ms-2"
                >
                  Change
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <button
      class="d-flex justify-content-center mt-4 bg-light-blue border-0 btn btn-primary"
      type="button"
    >
      Load More
    </button>
  </div>
</div>
`;
